src/nni_manager/training_service/kubernetes/kubernetesTrainingService.ts

```diff
@@ -357,6 +358,50 @@ abstract class KubernetesTrainingService {
         );
         return registrySecretName;
     }
+
+    protected async uploadFilesToAzureStorage(trialJobId: string, trialLocalTempFolder: String, codeDir: String): Promise<string> {
+        if (this.azureStorageClient === undefined) {
+            throw new Error('azureStorageClient is not initialized');
+        }
+        let trialJobOutputUrl: string = '';
+        let retryCount: number = 1;
+        let config = yaml.safeLoad(fs.readFileSync('./config/config.yaml', 'utf8'));
+        if(config && config.azureStorageUploadRetryCount instanceof Number) {
+            retryCount = config.azureStorageUploadRetryCount;
+        }
+        let resultUploadNNIScript: boolean = false;
+        let resultUploadCodeFile: boolean = false;
+        try {
+            do {
+                //upload local files, including scripts for running the trial and configuration (e.g., hyperparameters) for the trial, to azure storage
+                if(!resultUploadNNIScript) {
+                    resultUploadNNIScript = await AzureStorageClientUtility.uploadDirectory(this.azureStorageClient,
+                        `nni/${getExperimentId()}/${trialJobId}`, this.azureStorageShare,
+                        `${trialLocalTempFolder}`);
+                }
+                //upload code files to azure storage
+                if(!resultUploadCodeFile) {
+                    resultUploadCodeFile = await AzureStorageClientUtility.uploadDirectory(this.azureStorageClient,
+                        `nni/${getExperimentId()}/${trialJobId}`, this.azureStorageShare,
+                        `${codeDir}`);
+                }
+                if (resultUploadNNIScript && resultUploadCodeFile) {
+                    trialJobOutputUrl = `https://${this.azureStorageAccountName}.file.core.windows.net/${this.azureStorageShare}` + 
+                    `/${path.join('nni', getExperimentId(), trialJobId, 'output')}`;
+                    break;
+                } else {
+                    //wait for 5 seconds to re-upload files
+                    await delay(5000);
+                    this.log.info('Upload failed, Retry: upload files to azure-storage');
+                }
+            } while (retryCount-- >= 0)
+        } catch (error) {
+            this.log.error(error);
+            //return a empty url when got error
+            return Promise.resolve("");
+        }
+        return Promise.resolve(trialJobOutputUrl);
```

Comment:
after try `retryCount` times, the files are still not uploaded, what does this function return?

Reply:
this function will return an empty path, and submitTrial function will set trial status FAILED.

Reply:
better to write log to tell the FAILED is induced by data upload.

Reply:
add log `Retry-count is used up, upload files to azureStorage for trial ${trialJobId} failed!`